merkle/crdt/composite.go

```diff
@@ -64,25 +32,20 @@ type MerkleCompositeDAG struct {
 // NewMerkleCompositeDAG creates a new instance (or loaded from DB) of a MerkleCRDT
 // backed by a CompositeDAG CRDT.
 func NewMerkleCompositeDAG(
-	datastore datastore.DSReaderWriter,
-	headstore datastore.DSReaderWriter,
-	dagstore datastore.DAGStore,
+	txn datastore.Txn,
```

Comment:
suggestion: (This is changed in a few places, so I am just commenting here and applying the thought everywhere it has changed) I don't think we should be using `Txn` type here. Its a rather broad and overly powerful type to expose into the CRDT system, which was previously and purposefully simple in its dependencies, only specifying that it needed a datastore, headstore, and dagstore, which each of which could potentially come from different places, and only needed to implement the base `DSReaderWriter` (and `DAGStore`) interfaces.

We can probably drop the individual params as you are doing now, but it should be opened up to the entirety of the `Txn` type.

My suggestion would be to define a interface within the merklecrdt package that is a slimmed down version of the Multistore, specifically one that exposes 3 methods (`Datastore()`, `Headstore()`, and `DAGStore()`).

Reply:
I see what you mean. I'll make the change 👍.

Reply:
✅